src/Poseidon/Package.hs

```diff
@@ -280,3 +330,62 @@ compFunc1 (EigenstratSnpEntry c1 p1 _ _ _ _, _) (EigenstratSnpEntry c2 p2 _ _ _
 compFunc2 :: (EigenstratSnpEntry, GenoLine) -> [(EigenstratSnpEntry, GenoLine)] -> Ordering
 compFunc2 (EigenstratSnpEntry c1 p1 _ _ _ _, _) ((EigenstratSnpEntry c2 p2 _ _ _ _, _):_) = compare (c1, p1) (c2, p2)
 compFunc2 _                                     []                                        = error "compFunc2 - should never happen"
+
+-- Janno file loading
+
+decodingOptions :: Csv.DecodeOptions
+decodingOptions = Csv.defaultDecodeOptions { 
+    Csv.decDelimiter = fromIntegral (ord '\t')
+}
+
+instance Csv.FromRecord PoseidonSample
+
+-- | A helper function for parsing double lists
+bytestringToDouble :: [Bch.ByteString] -> [Double]
+bytestringToDouble [] = []
+bytestringToDouble (x:xs) = (read (Bch.unpack x) :: Double) : bytestringToDouble xs
+
+instance Csv.FromField [Double] where
+    parseField = fmap bytestringToDouble . fmap (\x -> Bch.splitWith (==';') x) . Csv.parseField
+
+-- | A helper function for parsing integer lists
+bytestringToInteger :: [Bch.ByteString] -> [Integer]
+bytestringToInteger [] = []
+bytestringToInteger (x:xs) = (read (Bch.unpack x) :: Integer) : bytestringToInteger xs
```

Comment:
Same here... maybe the error gets somehow propagated to the general CSV reading routine and things are fine. In any case this wouldn't prevent us from merging this PR and fixing later, but it's something to keep an eye on. We anyway need some automatic tests for Janno parsing.